securesignals-ima/src/main/java/com/uid2/securesignals/ima/UID2SecureSignalsAdapter.kt

```diff
@@ -41,7 +41,10 @@ public class UID2SecureSignalsAdapter : SecureSignalsAdapter {
             UID2Manager.init(context)
         }
 
-        callback.onSuccess()
+        // After we've asked to initialize the manager, we should wait until it's complete before reporting success.
+        // This will potentially allow any previously persisted identity to be fully restored before we allow any
+        // signals to be collected.
+        UID2Manager.getInstance().onInitialized = callback::onSuccess
```

Comment:
We now report `onSuccess ` **after** we've initialized, rather than after we've **started** initializing.